api/src/main/java/org/eclipse/microprofile/lra/annotation/ws/rs/LRA.java

```diff
@@ -215,12 +215,59 @@
          *     to the caller.
          * </p>
          */
-        NEVER
+        NEVER,
+
+        /**
+         * <p>
+         *     An activity can be scoped within an existing LRA using the Nested
+         *     element value. A new LRA will be created even if there is already
+         *     once present when the method is invoked, i.e. these LRAs will then
+         *     either be top-level or nested automatically depending upon the
+         *     context within which they are created. If invoked without a context
+         *     the new LRA will be top level. If invoked with an LRA present a
+         *     new nested LRA is started whose outcome depends upon whether the
+         *     enclosing LRA is closed or cancelled:
+         * </p>
+         *
+         * <ul>
+         *     <li>If the nested LRA is closed but the outer LRA is cancelled then
+         *     the participants registered with the nested LRA will be told to compensate.
+         *     </li>
+         *     <li>If the nested LRA is cancelled the outer LRA can be still closed
+         *     (and in this case the nested participants will not be asked to complete).
+         *     </li>
```

Comment:
We should mention explicitly that the participant will gets called twice, once for the nested LRA, and once for the enclosing LRA. (only when called with an enclosing LRA of course, not when top level)

Reply:
But the parent and nested LRA are two different LRAs so the participant will be called with a different context.

Reply:
exactly. But JAX-RS resource is unaware of the value of the enclosing LRA id (or should we add it as another header to the method annotated with @NestedLRA)
Otherwise the @Complete and @Compensate methods are called with a LRAid which is unknown to the JAX-RS resource (and methods) and they don't know what to do with it.

Reply:
If the enclosing LRA was created elsewhere and the current resource method is marked NESTED then the resource will only be enlisted with the NESTED LRA, hence there is no ambiguity.

But I agree that adding another header to identify the parent LRA is needed in order to support the nesting semantics I have outlined in a later comment. If I add that I can add text that will explain better how the callbacks work for nesting.

Reply:
So I added the extra explanatory test you asked for and I also added the extra JAX-RS header.